### ISSUE TITLE
Creation of auto_increment columns requires manual corrections to work with MySQL/MariaDB

### ISSUE THREAD

**Issue by BloodyRain2k** [dbeaver/dbeaver#21998]
### Description

I've started using DBeaver because I needed a replacement for PhpMyAdmin, and for the most part has it been a good replacement.
But creating new columns for the MySQL db I managed over PMA before, requires a couple additional manual steps that I think shouldn't be needed.

The biggest issue is that DBeaver puts the `NULL|NOT NULL` portion of a column creation query at the end of that column's code, and MySQL apparently doesn't like that.
This is the column configuration I made with the dialog:
![Screenshot from 2023-11-26 15-33-16](https://github.com/dbeaver/dbeaver/assets/1014362/c956eb65-7254-42ab-97f8-73bc1621b961)

This is the resulting SQL:
![Screenshot from 2023-11-26 15-36-38](https://github.com/dbeaver/dbeaver/assets/1014362/07ad49ec-e66e-4889-82e0-cccafda2da4d)

And this is the resulting error:
![Screenshot from 2023-11-26 15-57-40](https://github.com/dbeaver/dbeaver/assets/1014362/69cfeb6e-1862-4442-b6dc-9aadac23b11a)

I literally didn't even understand why it's complaining, so I recreated the same in PMA and the only difference I noticed, aside of the lacking quotation marks, was that `NULL|NOT NULL` always followed directly after the column's type.
So I just edited the SQL DBeaver gave me like this and it worked right away.
![Screenshot from 2023-11-26 15-33-58](https://github.com/dbeaver/dbeaver/assets/1014362/58d3d3cc-d30e-4faa-9713-3db85fd91cfe)

A few minor issues, which are more QoL, are that there's no checkbox for "primary key", for which I also had to refer to PMA again to figure out how to get the SQL result correctly. (no points for correctly guessing that I rarely work with databases)
Would be nice if that could also have a checkbox like "auto increment" has. Or if it could just add `,primary key` to "Extra" if "auto increment" is checked and the database is MariaDB.
Because without the PK, it complains that there's only one column with "auto increment" allowed and it needs to be a key:
![Screenshot from 2023-11-26 16-07-06](https://github.com/dbeaver/dbeaver/assets/1014362/6d184c42-2523-4219-8c7c-1842bab267db)

The other thing is that checking "auto increment" doesn't change the column type from `varchar(100)` to an `INT`, so without also changing that, the first complaint will be about the column type:
![Screenshot from 2023-11-26 17-11-58](https://github.com/dbeaver/dbeaver/assets/1014362/b7a12f34-420b-4bec-96eb-0ccb09066d9c)

### DBeaver Version

Community 23.2.5.202311191730

### Operating System

Debian GNU/Linux 12 bookworm / 6.1.0-13-amd64

### Database and driver

DB version: 10.4.28-MariaDB
Driver: MySQL

### Steps to reproduce

connect to the DB that (Xammp) MySQL created.
create a Database with default settings: `utf8mb4` & `utf8mb4_general_ci`
create table / add column to existing table with "auto increment"
get every error mentioned above

### Additional context

The DB runs on MySQL on a different PC running Win 10, and while I don't think that should be relevant, I'll just point it out anyways.
I'm also rarely working with databases, so there's a high chance I missed providing something needed, in which case just tell me what and where I find that info and I'll add it.

**Comment by LonwoLonwo** [comment]
Hello @BloodyRain2k 

So, you added primary key info manually in the extra field? And it is not working this way?

How about using a special command for key creation instead?

![2023-11-30 11_11_01-DBeaver Ultimate 23 3 0 - NewTable_1](https://github.com/dbeaver/dbeaver/assets/45152336/0e5dbeeb-8a22-46ab-a0e1-37d156ecaf8a)

In this case, the result DDL will be correct.

![2023-11-30 11_11_16-NewTable_1 - Persist Changes](https://github.com/dbeaver/dbeaver/assets/45152336/59473580-39d5-41c1-9f99-924677f5ca77)

**Comment by BloodyRain2k** [comment]
@LonwoLonwo yes, that approach works, leaving aside that I still have to manually correct the column type to INT.

I'm still unsure why the "create column" dialog has an option for "Auto Increment" when the user still has to correct the configuration for that afterwards.

As it is right now:
it does *not* change the column type to something compatible with "auto_increment"
it does *not* automatically set the column as a "primary key", which at least in MariaDB's case is mandatory
it puts the "extra" data between the column type and "NOT NULL", which again in MariaDB's case seems to be an issue, with *only* "auto_increment" being an exception apparently

**Comment by E1izabeth** [comment]
We don't create foreign keys automatically, because not every column with auto-increment is a key.

**Comment by E1izabeth** [comment]
Thank you for bug report.
